### DISCUSSION TITLE
`dotnet restore` commands possible when solution contains WiX v3 action projects?

### DISCUSSION THREAD

**Discussion by wnewbery** [wixtoolset/issues#7958] [Questions]
While trying to update our version of Paket (so a new project can target .NET 8), we found it wants to run `dotnet restore example.sln`, but this is failing when some projects are using WiX 3.11.2.3516 which appears to be the latest V3 version. We are not looking to update the the WiX projects themselves, and they have `<TargetFrameworkVersion>v4.8</TargetFrameworkVersion>`.

```
C:\Program Files (x86)\MSBuild\Microsoft\WiX\v3.x\Wix.CA.targets(42,5): error MSB4062: The "ReadRegistry" task
could not be loaded from the assembly C:\Program Files (x86)\WiX Toolset v3.11\bin\WixTasks.dll. Could not load
 file or assembly 'Microsoft.Build.Utilities, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the ass
embly and all its dependencies are available, and that the task contains a public class that implements Microso
ft.Build.Framework.ITask. [C:\Demo\Wix3CustomActions\Wix3CustomActions.csproj]
```

I think this error comes from the project having `<WixCATargetsPath Condition=" '$(WixCATargetsPath)' == '' ">C:\Program Files (x86)\MSBuild\Microsoft\WiX\v3.x\Wix.CA.targets</WixCATargetsPath>` and `<Import Project="$(WixCATargetsPath)" />`.

Is it possible to use the new `dotnet restore` tools with WiX 3 or do we need to upgrade to WiX 4 first?

Could the WiX targets files maybe skip the `ReadRegistry` bit on a `dotnet restore`, as I assume they never produce anything relevant to restoring dependencies, and the WiX build tools are installed separately anyway? Some `Condition` on the `<Import />` to only import WiX on msbuild maybe?

https://github.com/wixtoolset/issues/issues/5711 seems to imply removal of Microsoft.Build.Utilities 2.0 but that version has never been released?

https://github.com/wixtoolset/issues/issues/5627 seems to imply `dotnet build` works in WiX 4, we don't need `dotnet build` to work, as we build from either VS2022 or msbuild.

**Comment by robmen** [comment]
WiX v3 does not support `dotnet.exe` commands. You must move to WiX v4 for that.
